src/App/pages/productDetailPage/ProductDetailPage.tsx

```diff
@@ -0,0 +1,108 @@
+import { FC, useState, useEffect } from "react";
+
+import Button from "@components/Button";
+import { ButtonColor } from "@components/Button";
+import WithLoader from "@components/WithLoader";
+import { PRODUCTS } from "@config/apiUrls";
+import { Product } from "@config/types";
+import Navbar from "@pages/productsPage/components/Header/components/Navbar";
+import ProductList from "@pages/productsPage/components/ProductList";
+import axios from "axios";
+import { useParams } from "react-router-dom";
+
+import style from "./ProductDetailPage.module.scss";
+
+const initialValue: Product = {
+  id: 0,
+  title: "",
+  description: "",
+  price: 0,
+  image: "",
+  category: "",
+  rating: {
+    count: 0,
+    rate: 0,
+  },
+};
+
+const ProductDetailPage: FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [{ image, title, description, price, category }, setProduct] =
+    useState<Product>(initialValue);
```

Comment:
тут логично, если 
```suggestion
  const [product, setProduct] =
    useState<Product | null>(null);
```

И ниже смотрим если `loading` - показываем лоадер, если нет и есть`product` есть показываем продукт, если все `false`, получается ошибка при запросе за данными